tests/translator/input/error_gateway_response_invalid_type_int.yaml

```diff
@@ -0,0 +1,19 @@
+Resources:
+  ApiResource:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: Prod
+      GatewayResponses:
```

Comment:
I am not following how this results in a "type int" failure. Can you walk me through it so I can understand?

Reply:
`GatewayResponses` expects a `GatewayResponse` object. Here, you can see that there is a provided `StatusCode` field which is an `int`. Maybe a bit unclear in this test now that you mention it, but this is a very similar case to some the customer template issues we've been seeing.

Reply:
I see. Error is a little confusing to me but I don't have a better recommendation at the moment. Raising is also better then SAM silently failing so let's start here and see if the community has a better messaging.